```
>>> import numpy as np
>>> a = np.arange(9.).reshape(3, 3)
>>> v = np.arange(1., 4.)
2次元配列
>>> a
array([[0., 1., 2.],
       [3., 4., 5.],
       [6., 7., 8.]])
1次元配列
>>> v
array([1., 2., 3.])
積を求める（１次元配列の方向は自動解釈される）
>>> np.dot(a, v)
array([ 8., 26., 44.])
引数を逆にする（１次元配列の方向は自動解釈される）
>>> np.dot(v, a)
array([24., 30., 36.])
形状を(3, 1)の2次元配列にする
>>> u = v.reshape(-1, 1)
>>> u
array([[1.],
       [2.],
       [3.]])
>>> np.dot(a, u)
array([[ 8.],
       [26.],
       [44.]])
引数を逆にするとエラー
>>> np.dot(u, a)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: shapes (3,1) and (3,3) not aligned: 1 (dim 1) != 3 (dim 0)
形状を(1, 3)の2次元配列にする
>>> w = v.reshape(1, -1)
>>> w
array([[1., 2., 3.]])
>>> np.dot(w, a)
array([[24., 30., 36.]])
```
